EDSC-2672: Implement lazy loading on timeline container

sofiarivas opened the pull request:
# Overview
This PR addresses this issue: https://github.com/nasa/earthdata-search/issues/1219

### What is the feature?
- Implements lazy loading Timeline Container component

### What is the Solution?
- It uses React `lazy` and `Suspense` to handle lazy loading the timeline container component. 
- Adds the `Spinner` component as the fallback to display as the code for the timeline is being loaded.

### What areas of the application does this impact?
- The search feature

# Testing

### Reproduction steps

- **Environment for testing:**
This can be tested locally

- **Collection to test with:**

1. Run the application
2. You can test this with the usage of the browser dev tools. 
- To test the fallback component: You can emulate a slow network in order to make it easier. This can be done on the network panel of your browser devtools. Then click one of the search results and validate that on the bottom of the page you see the Spinner. After the JS file has been loaded validate you see the timeline and it should work as expected.
- To test the code splitting: On the network panel, you should see multiple js bundles been loaded after you click on the search results. These files contain the code for the timeline component which is being lazy-loaded.

### Attachments

Lazy loading the timeline component:

![image](https://user-images.githubusercontent.com/13956453/95030288-72fba700-0674-11eb-97d3-613103868769.png)



# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

mreese84 commented:
Please make sure that you're running the tests (`npm run test` is the command, you can find other information in the readme) -- your PR has a failing test on some of the changes you made.

Also, the particular dataset that you selected for your screenshot/gif is not the best data type to use for this. Anything with that gold "CWIC" badge is _technically_ not NASA data, it's data from international partners. Data availability (ie the granules timeline are trying to load) does not have the same standards that we have/control for our data. I would recommend searching for `C1621383370-LPDAAC_ECS` and using that to ensure granules are coming back and populating the timeline correctly in any tests that you run.

sofiarivas commented:
Thanks, @mreese84 sorry I didn't upload the unit tests fix before submitting the PR. But on my latest commit I have fixed them and I also updated the screenshot to show how the timeline is being populated. 

Let me know if there is anything else needed to get a proper review 🙂

abbottry commented on static/src/js/containers/TimelineContainer/TimelineContainer.js:
```diff
@@ -71,18 +73,25 @@ export const TimelineContainer = (props) => {
-      onMetricsTimeline={onMetricsTimeline}
-    />
+    <Suspense fallback={(
+      <div className="timeline-loading">
```
Given the UI and 'data' that this is loading, we think it's safe to remove any loading indicators for this PR. So, please remove the fallback attribute here and all corresponding code.

sofiarivas replied:
Sounds good. I have replaced it with a null value since the fallback property can't be omitted from the Suspense component.

abbottry commented on static/src/js/containers/TimelineContainer/TimelineContainer.scss:
```diff
@@ -0,0 +1,7 @@
+.timeline-loading {
```
No longer necessary, please remove.

sofiarivas replied:
Done, removed on the latest commit https://github.com/nasa/earthdata-search/pull/1235/commits/3be40675fd09ea6f1a3dac52d2bb4096b6040599

mreese84 commented:
Thank you for your contribution @sofiarivas! This will be in our 1.130.x release which is scheduled for deployment to our Prod environment no later than 10/28/20.

[search.earthdata.nasa.gov](http://search.earthdata.nasa.gov)
